### PR TITLE
Run coin selection properties with non-zero withdrawal

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

From looking at  #2006, but surely not related.

# Overview

- [x] I added a withdrawal field to `CoinSelProp` and used it in coin selection properties

# Comments

- I'm not that familiar with the coin selection, but this seems like a good idea, right?

### CODE REVIEW

File: lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelectionSpec.hs [hunk 1]
```diff
@@ -339,9 +346,17 @@ instance Arbitrary a => Arbitrary (NonEmpty a) where
         NE.fromList <$> vector n
 
 instance Arbitrary CoinSelProp where
-    shrink (CoinSelProp utxo outs) = uncurry CoinSelProp
-        <$> zip (shrink utxo) (shrink outs)
-    arbitrary = applyArbitrary2 CoinSelProp
+    shrink (CoinSelProp utxo wdrl outs) =
+        tail $ CoinSelProp <$> shrinkUTxO <*> shrinkWithdrawal <*> shrinkOuts
```
Comment (Anviking) [inline lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelectionSpec.hs]: Think list comprehensions are more idiomatic

File: lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelectionSpec.hs [hunk 2]
```diff
@@ -339,9 +345,17 @@ instance Arbitrary a => Arbitrary (NonEmpty a) where
...
-    arbitrary = applyArbitrary2 CoinSelProp
+    shrink (CoinSelProp utxo wdrl outs) =
+        [ CoinSelProp utxo' wdrl outs | utxo' <- shrink utxo ]
+        ++ [ CoinSelProp utxo wdrl' outs | wdrl' <- shrinkWdrl wdrl ]
+        ++ [ CoinSelProp utxo wdrl outs' | outs' <- shrink outs ]
+      where
+        shrinkWdrl = map Quantity . shrink . getQuantity
+    arbitrary = do
+        utxo <- arbitrary
+        wdrl <- Quantity <$> frequency [(65, return 0), (35, arbitrary)]
```
Comment (Anviking) [inline lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelectionSpec.hs:356]: 65% no withdrawal like previously

File: lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelectionSpec.hs [hunk 3]
```diff
@@ -227,14 +230,17 @@ data CoinSelectionsSetup = CoinSelectionsSetup
 data CoinSelProp = CoinSelProp
     { csUtxO :: UTxO
         -- ^ Available UTxO for the selection
+    , csWithdrawal :: Quantity "lovelace" Word64
+    -- ^ Availible Withdrawal
```
Comment (paweljakubas) [inline lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelectionSpec.hs]:
```suggestion
    -- ^ Available Withdrawal
```

**Review by paweljakubas — APPROVED** [review]
LGTM I wonder if we can add some properties, especially when withdrawal is edge case in comparison with inputs :thinking:

**Pull request merged**